src/Common.cpp

```diff
@@ -228,11 +228,13 @@ bool NAS2D::endsWith(const std::string& string, const std::string& end) noexcept
 
 bool NAS2D::startsWith(const std::string& string, char start)
 {
+	if (string.empty()) { return false; }
 	return *std::begin(string) == start;
 }
 
 bool NAS2D::endsWith(const std::string& string, char end)
 {
+	if (string.empty()) { return false; }
 	return *(std::end(string) - 1) == end;
```

Comment:
Similar to the above, but also using [`string::back()`](https://en.cppreference.com/w/cpp/string/basic_string/back) in place of `string::end() - 1`, this could be written as:
```cpp
return !string.empty() && (*string.back() == end);
```

I don't believe there is a `std::back` method.

Edit: I see you caught that `back()` method already.